src/DocRaptor.php

```diff
@@ -39,15 +40,17 @@ class ApiWrapper
     protected $baseurl;
 
 
-
     /**
+     * @param HttpTransferInterface $httpClient
      * @param string|null $api_key
      */
-    public function __construct($api_key = null)
+    public function __construct(HttpTransferInterface $httpClient, $api_key = null)
```

Comment:
Are classes not objects in PHP? My thinking is informed by ruby DI, which is a little different. Is default params a thing? Like could our config object just be an associative array, and then we set anything we didn't get *but* have known good defaults for (i.e. HttpClient).

I'm going for extreme ease of use for consumers of the class.